nexus/src/db/subnet_allocation.rs

```diff
@@ -490,142 +305,585 @@ impl QueryFragment<Pg> for FilterConflictingVpcSubnetRangesQueryValues {
     }
 }
 
+/// Errors related to inserting or attaching a NetworkInterface
+#[derive(Debug)]
+pub enum NetworkInterfaceError {
+    /// The instance specified at create or attach is associated with a
+    /// different VPC
```

Comment:
different VPC from what?  Some other interface that's already attached?

Reply:
Yes, that's right. I can update the comment, but this is catching the case where an instance has one or more NICs from VPC A, and the NIC to be attached in this query is from VPC B, where A != B.